File: MeetCode.Domain/Entities/Problem.cs
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;

namespace MeetCode.Domain.Entities;

public partial class Problem
{
    public Guid ProblemId { get; set; }

    public string Slug { get; set; } = null!;

    public string Title { get; set; } = null!;

    public string StatementMd { get; set; } = null!;

    public string Difficulty { get; set; } = null!;

    public int TimeLimitMs { get; set; } = 1;

    public int MemoryLimitMb { get; set; } = 1;

    public Guid CreatedBy { get; set; }

    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset? UpdatedAt { get; set; }

    public int TotalSubmissionCount { get; set; } = 0;

    public int ScoreAcceptedCount { get; set; } = 0;

    public double? AcceptanceRate { get; set; }

    public bool IsActive { get; set; }

    public virtual User CreatedByNavigation { get; set; } = null!;

    public virtual ICollection<Submission> Submissions { get; set; } = new List<Submission>();

    public virtual ICollection<TestCase> TestCases { get; set; } = new List<TestCase>();

    public virtual ICollection<ProblemTag> Tags { get; set; } = new List<ProblemTag>();
    public virtual ICollection<ProblemTemplate> ProblemTemplates { get; set; } = new List<ProblemTemplate>();

    public override string ToString()
    {
        return this.ToGenericString();
    }

    public void GenerateSlug()
    {
        Slug = Regex.Replace(Title.ToLowerInvariant().Trim(), @"\s+", "-").ToLowerInvariant();
    }

    public void UpdateSubmission(string submissionVerdict)
    {
        TotalSubmissionCount++;
        if (submissionVerdict == "accepted")
        {
            ScoreAcceptedCount++;
        }
        AcceptanceRate = ScoreAcceptedCount / TotalSubmissionCount;
    }

    public void ToggleStatus()
    {
        IsActive = !IsActive;
    }

    public void UpdateBasic(string statementMd, string difficulty)
    {
        StatementMd = statementMd;
        Difficulty = difficulty;
    }

    public void UpdateTags(IEnumerable<ProblemTag> tagList)
    {
        Tags.Clear();
        foreach(var tag in tagList)
        {
            Tags.Add(tag);
        }
    }
}

File: meetcode.client/src/types/admin/problem.ts
import { ProblemTag } from "./problemTag";
import { ProblemTemplate } from "./problemTemplate";
import { Submission } from "./submission";
import { TestCase } from "./testCase";

export type Difficulty = 'easy' | 'medium' | 'hard';

export const difficultyStyle: Record<Difficulty, string> = {
  easy: "text-emerald-300",
  medium: "text-yellow-300",
  hard: "text-red-300",
};

export interface Problem {
    problemId: string;
    slug: string;
    title: string;
    statementMd: string;
    difficulty: Difficulty;
    timeLimitMs: number;
    memoryLimitMb: number;
    createdBy: string;
    createdAt: string;
    updatedAt: string | null;
    totalSubmissionCount: number;
    scoreAcceptedCount: number;
    acceptanceRate: number | null;
    isActive: boolean;
    submissions: Array<Submission>;
    testCases: Array<TestCase>;
    tags: Array<ProblemTag>;
    problemTemplates: Array<ProblemTemplate>;
}